app/controllers/email_alert_subscriptions_controller.rb

```diff
@@ -5,16 +5,25 @@ class EmailAlertSubscriptionsController < ApplicationController
   helper_method :subscriber_list_params
 
   def create
-    if valid_choices?
-      redirect_to email_alert_signup_api.signup_url
-    else
-      @error_message = "Please choose an email alert"
-      render action: :new
-    end
+    validate_choices
+    redirect_to email_alert_signup_api.signup_url
+  rescue MissingFiltersError
+    render_error "Please choose an email alert"
+  rescue EmailAlertSignupAPI::UnprocessableSubscriberListError
+    render_error "Unprocessable request. Please check your filters and try again."
```

Comment:
Is this message displayed to users?  If it is, I don't think "unprocessable request" is a very useful message to give.  Even "An error occurred" is probably better.